src/ansys/fluent/core/filereader/casereader.py

```diff
@@ -73,36 +76,53 @@ class CaseReader:
         Get the precision (1 or 2 for 1D of 2D)
     """
 
-    def __init__(self, hdf5_case_filepath: str):
+    def __init__(self, case_filepath: str):
+
         try:
-            file = h5py.File(hdf5_case_filepath)
+            if Path(case_filepath).suffix == ".h5":
+                file = h5py.File(case_filepath)
+                settings = file["settings"]
+                rpvars = settings["Rampant Variables"][0]
+                rp_vars_str = rpvars.decode()
+            elif Path(case_filepath).suffix == ".cas":
+                with open(case_filepath, "rb") as file:
+                    rp_vars_str = file.read()
+                rp_vars_str = get_processed_string(rp_vars_str)
+
+            elif Path(case_filepath).suffix == ".gz":
```

Comment:
We could check for the full extension, `.cas.gz`

Reply:
Done